tardis/__init__.py

```diff
@@ -33,6 +33,122 @@
+    to a particular level
+
+    Parameters
+    ----------
+    log_level : logging object
+        allows to have a filter for the
+        particular log_level
+    """
+
+    def __init__(self, log_level):
+        self.log_level = log_level
+
+    def filter(self, log_record):
+        """
+        filter() allows to set the logging level for
+        all the record that are being parsed & hence remove those
+        which are not of the particular level
+
+        Parameters
+        ----------
+        log_record : logging.record
+            which the paricular record upon which the
+            filter will be applied
+
+        Returns
+        -------
+        boolean : True, if the current log_record has the
+            level that of the specified log_level
+            False, if the current log_record doesn't have the
+            same log_level as the specified one
+        """
+        return log_record.levelno == self.log_level
+
+
+# Setting up a list to store the Logging Filters set by logger.addFilter()
+list_of_filter = []
+
+
+def logging_state(log_state, tardis_config, specific):
+    """
+    Function to set the logging configuration for the simulation output
+    Called from within run_tardis()
+    Configured via functional arguments passed through run_tardis() - log_state & specific
+    Configured via YAML parameters under `debug` section - logging_level & specific_logging
+
+    Parameters
+    ----------
+    log_state: str
+        Allows to input the log level for the simulation
+        Uses Python logging framework to determine the messages that will be output
+    specific: boolean
+        Allows to set specific logging levels. Logs of the `log_state` level would be output.
+    """
+    if "debug" in tardis_config:
+        if specific or tardis_config["debug"]["specific"]:
+            specific = True
+        else:
+            specific = False
+
+        if tardis_config["debug"]["log_state"] or log_state:
+            if (
+                log_state.upper() == "CRITICAL"
+                and tardis_config["debug"]["log_state"]
+            ):
+                logging_level = tardis_config["debug"]["log_state"]
+            elif log_state:
+                logging_level = log_state
+                if tardis_config["debug"]["log_state"] and log_state:
+                    print(
+                        "log_state is defined both in Functional Argument & YAML Configuration {debug section}"
+                    )
+                    print(
+                        f"log_state = {log_state} will be used for Log Level Determination\n"
+                    )
+            else:
+                logging_level = tardis_config["debug"]["log_state"]
+    else:
+        logging_level = log_state
+        specific = specific
+
+    loggers = [
+        logging.getLogger(name) for name in logging.root.manager.loggerDict
+    ]
+    if logging_level.upper() in LOGGING_LEVELS.keys():
```

Comment:
This conditional passes without error if the specified log level is *not* in the LOGGING_LEVELS dict.  There should be an error raised if a user gives an invalid input.  Similarly, something should happen in line 142.

The error message that gets returned should enumerate the valid logging states for the user.

Reply:
Yes, this is a nice thing to have 😄. I have added an `Value Error` which is raised whenever the user doesn't enter a valid value to the `log_state` parameter. 
For the `specific` parameter, I don't think an exception would be needed as it only has boolean states. Log levels are managed by the `log_state` parameter & hence, raising the above exception would suffice for this case as well 😉